resources/splunk_app.rb

```diff
@@ -33,11 +31,11 @@ def app_path
   end
 
   def config_scope
-    'local'
+    @config_scope ||= (resource_name == :splunk_app_custom ? 'default' : 'local')
   end
 
-  def splunk_app_path
-    Pathname.new(install_dir).join('etc/apps')
+  def app_cache_path
+    @app_cache_path ||= Pathname.new(Chef::Config['file_cache_path']).join('splunk_ingredient/old_apps')
```

Comment:
Simple references would be something like `app_path + 'local'`
The case where I want to be locally consistent are like `Pathname.new('/opt/splunk').join('etc')` as opposed to `Pathname.new('/opt/splunk/') + 'etc'`
And cases where methods are being chained from the pathname object would almost require using join `app_path.join('local').mkpath` (I suppose one could do `(app_path + 'local').mkpath` but I'm not aware if thats stylistically acceptable)

There are some varying cases throughout the entire project... I would like to clean them up, but I don't entirely feel this is the time to do that. I will make an effort to make pathname joins consistent in the next PR.

I would like to continue this conversation as well, as the consensus might be to prefer using `+` or `.join` everywhere.